openstack/messaging/v2/messages/requests.go

```diff
@@ -103,3 +103,38 @@ func Create(client *gophercloud.ServiceClient, queueName string, opts CreateOpts
 	})
 	return
 }
+
+// DeleteMessageOptsBuilder allows extensions to add additional parameters to the
+// DeleteMessages request.
+type DeleteMessageOptsBuilder interface {
+	ToMessagesDeleteQuery() (string, error)
+}
+
+// DeleteMessageOpts params to be used with DeleteMessages.
+type DeleteMessageOpts struct {
+	Ids []string `q:"ids,omitempty"`
+	Pop int      `q:"pop,omitempty"`
+}
+
+// ToMessagesDeleteQuery formats a DeleteMessageOpts structure into a query string.
+func (opts DeleteMessageOpts) ToMessagesDeleteQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// DeleteMessages deletes multiple messages based off of ID or number of messages.
+func DeleteMessages(client *gophercloud.ServiceClient, queueName string, opts DeleteMessageOptsBuilder) (r DeleteResult) {
```

Comment:
This could also be named `Delete` since "messages" is the focus of this package.

But if there will be future functions in this package that would make `Delete` ambiguous and confusing, then `DeleteMessages` is fine.

Reply:
Yeah, thats why I had the distinction. Later there will be a "DeleteMessage" function.